File: production-order-vite/src/types/productionOrder.ts
export interface ProductionOrder {
  id: number;
  orderNumber: string;
  productCode: string;
  quantityPlanned: number;
  quantityProduced: number;
  status: number;
  startDate: string;
  endDate?: string;
  product?: {
    description: string;
  };
}

export interface CreateProductionOrderDto {
  orderNumber: string;
  productCode: string;
  quantityPlanned: number;
  status: number;
  startDate: string;
}

export interface UpdateProductionOrderDto {
  orderNumber?: string;
  productCode?: string;
  quantityPlanned?: number;
  status?: number;
  startDate?: string;
  endDate?: string;
}

export interface Product {
  id: number;
  code: string;
  description: string;
}

File: ProductionOrderApi/Models/ProductionOrder.cs
using ProductionOrderApi.Enums;

namespace ProductionOrderApi.Models
{
    public class ProductionOrder
    {
        public int Id { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public string ProductCode { get; set; } = string.Empty;
        public int QuantityPlanned { get; set; }
        public int QuantityProduced { get; set; }
        public ProductionOrderStatusEnum Status { get; set; } = ProductionOrderStatusEnum.Planejada;
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        public virtual Product? Product { get; set; }
        public virtual ICollection<ProductionLog> ProductionLogs { get; set; } = new List<ProductionLog>();
    }
}
